src/posix/platform/hdlc_interface.cpp

```diff
@@ -183,34 +182,95 @@ otError HdlcInterface::SendFrame(const uint8_t *aFrame, uint16_t aLength)
 otError HdlcInterface::Write(const uint8_t *aFrame, uint16_t aLength)
 {
     otError error = OT_ERROR_NONE;
+#if !OPENTHREAD_POSIX_VIRTUAL_TIME
+    struct timeval end;
+    struct timeval now;
+    struct timeval timeout = {kMaxWaitTime / 1000, (kMaxWaitTime % 1000) * 1000};
+    ssize_t        rval;
+
+    otSysGetTime(&now);
+    timeradd(&now, &timeout, &end);
 
-#if OPENTHREAD_POSIX_VIRTUAL_TIME
-    otSimSendRadioSpinelWriteEvent(aFrame, aLength);
-#else
     while (aLength)
     {
-        ssize_t rval = write(mSockFd, aFrame, aLength);
-
-        if (rval > 0)
+        if (WaitForWritable(timeout))
```

Comment:
only wait if write returns errno `EAGAIN`?

Reply:
Changed it.